employee_mgmt/app/models/employee.rb

```diff
@@ -1,30 +1,60 @@
 class Employee < ActiveRecord::Base
 
-  def update_phoneno val
+ validates :first_name, presence:true
+  validates :last_name, presence: true
+  validates :phone,allow_blank: true, format: { with: /9[0-9]{9}/,message:"Must have exactly 10 digits number starting with 9"}
+ validates :salary, numericality: {greater_than: 0}
+ validates :email,uniqueness: true,allow_blank: true, format: { with: /[a-zA-Z]+@[a-z0-9A-Z]+.[.a-zA-Z]+/,message:"%{value}Invalid Email ID"}
+
+ after_validation(on: :create) do
```

Comment:
1) We dont use parenthesis`()` for small conditions so we just use `if self.middle_name`  
2) We dont need to call self `self.first_name`, simply `first_name` will suffice  
3) rather than explicitly defining the value `first_name=first_name.capitalize` we use the bang version to change the value `first_name.capitalize!` 
4)we use a single line syntax whenever possible like in this case `middle_name.capitalize! if middle_name`